src/net_processing.cpp

```diff
@@ -2221,6 +2230,8 @@ void PeerManagerImpl::ProcessHeadersMessage(CNode& pfrom, const Peer& peer,
  */
 void PeerManagerImpl::ProcessOrphanTx(std::set<uint256>& orphan_work_set)
 {
+    if (!m_mempool) return;
```

Comment:
Same

Reply:
Same as above, a `m_tx_relay` check is done in the caller of this function too. However, I am still leaving the check here, with the debug build modification.